### DISCUSSION TITLE
Setting query options when using invoke

### DISCUSSION THREAD

**Discussion by janvennemann** [blitz-js/blitz#3807] [Q&A] [answered]
Hey,

i'm using [`invoke`](https://blitzjs.com/docs/resolver-client-utilities#invoke) to run a query on a button click. This works just fine, however for this particular query i don't want the automatic query retry that used by default for all queries. With `useQuery` i would just pass the query options as a third parameter to set `{ retry: false }`. I couldn't find a way to do this with `invoke` though. Is there something i'm missing or is this not possible with `invoke`?

**Comment by beerose** [comment, marked as answer]
The `retry` thing is a react-query option (when using resolvers from the client with useQuery etc). Invoke calls a Blitz resolver directly, so there's no react-query involved. Under the hood, it's a simple HTTP client that uses `fetch`. Hence there shouldn't be any additional retries.

**Reply by janvennemann** [reply]
Thanks for the explanation, that makes total sense. I tried to reproduce it today and it behaves exactly as expected and doesn't use retries. Not sure what was going on when i asked this question 🤷
